driver/http/winhttp.c

```diff
@@ -0,0 +1,155 @@
+#include "winhttp.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <windows.h>
+#include <winhttp.h>
+
+#include <euicc/interface.h>
+#include <lpac/utils.h>
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(HINTERNET, WinHttpCloseHandle);
+
+#define utf8_to_wide(input, output)                                            \
+    const int output##n = MultiByteToWideChar(CP_UTF8, 0, input, -1, NULL, 0); \
+    wchar_t output[output##n];                                                 \
```

Comment:
Don't use VLA, it will prevent compiler optimization, bring stack overflow vulnerabilities and lack support (although it's a mandatory feature of C99). Maybe we need `-Wvla`.